### PR TITLE
Feature/swift pm

### PR DESCRIPTION
**Pull request by melekr**
- Add Swift Package Manager support
- Add Swift Package Manager dependencies
- Remove Cassette dependency
- Replace Cassette's Queue and file operations

### CODE REVIEW

File: Backtrace.podspec [hunk 1]
```diff
@@ -9,7 +9,7 @@
 Pod::Spec.new do |s|
 
   s.name         = "Backtrace"
-  s.version      = "1.7.5"
+  s.version      = "1.7.6"
```
Comment (konraddysput) [inline Backtrace.podspec:12]: can we bump it to 1.8.0 (minimum). If we have breaking changes, can we bump it to 2.0.0?

File: Examples/Common/Keys.m [hunk 2]
```diff
@@ -5,18 +5,22 @@ @implementation Keys
 + (NSString const *) universeName {
     // Set your universe name here - https://support.backtrace.io/hc/en-us/articles/360040516451-What-is-a-submission-url-
     // e.g.: yourteam
-    return @"yourteam";
+    return @"melektest";
```
Comment (konraddysput) [inline Examples/Common/Keys.m:8]: this shouldn't be here

File: Examples/Example-iOS-ObjC/AppDelegate.m [hunk 3]
```diff
@@ -35,7 +35,7 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     // sending NSException
     @try {
         NSArray *array = @[];
-        array[1]; // will throw exception
+        //array[1]; // will throw exception
```
Comment (konraddysput) [inline Examples/Example-iOS-ObjC/AppDelegate.m:38]: this should be here

File: Examples/Common/Keys.m (hunk 2 above)
Comment (melekr) [reply Examples/Common/Keys.m:8]: that and other things too!
I'll cleanup soon

File: Backtrace.podspec (hunk 1 above)
Comment (melekr) [reply Backtrace.podspec:12]: 2.0.0 sounds great

File: Examples/Common/Keys.m (hunk 2 above)
Comment (konraddysput) [reply Examples/Common/Keys.m]: Yeah but if we're announcing to have something to test and share we need to be sure something like that will be cleaned up.

File: Backtrace.podspec (hunk 1 above)
Comment (konraddysput) [reply Backtrace.podspec]: Because the library loosing support for iOS/mac/tv 10 I think we should go with 2.0.0

File: Podfile [hunk 4]
```diff
@@ -4,19 +4,17 @@ source 'https://cdn.cocoapods.org/'
 
 # Definitions
 def shared_pods
-    # Version 1.5.3 has client-side symbolication turned off by default, which we want. Please don't upgrade to 1.5.4, which enables it by default.
-    pod 'Backtrace-PLCrashReporter', '1.5.3' 
+    # Define shared CocoaPods here
```
Comment (konraddysput) [inline Podfile:7]: in podfile definition shouldn't we have information about libraries that we used to have?

File: scripts/test.sh [hunk 5]
```diff
@@ -8,4 +8,4 @@ fastlane ios tests
 fastlane mac tests
 fastlane tvos tests
 
-pod lib lint --verbose --allow-warnings --sources='https://cdn.cocoapods.org/'
+#pod lib lint --verbose --allow-warnings --sources='https://cdn.cocoapods.org/'
```
Comment (konraddysput) [inline scripts/test.sh:11]: can we remove dead code?

File: Backtrace-iOS/Resources/test.txt [hunk 6]
```diff
@@ -0,0 +1,199 @@
+Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin aliquam massa a metus tempus, et facilisis velit faucibus. Suspendisse eu est condimentum, elementum justo at, viverra nulla. Donec feugiat arcu mauris. Maecenas fermentum tincidunt ligula, at molestie augue ornare sit amet. Ut id metus finibus justo lacinia rhoncus mollis rhoncus dui. Donec et massa quis lorem hendrerit semper. Fusce arcu massa, rutrum non cursus quis, tempor vitae augue. Quisque vestibulum ac ipsum nec aliquam. Nunc sem leo, blandit eget sodales id, sodales sed tellus. Fusce bibendum libero lacus, ut fringilla nunc dictum rhoncus. Aliquam lacus erat, imperdiet et euismod eget, accumsan in tellus. Nunc vel sapien id sem blandit faucibus. Curabitur mollis consequat lorem, ac fermentum libero vestibulum a. Aenean interdum pretium nisi, vitae semper nisi ultrices at. Nam dignissim magna non velit varius, sed ultrices turpis rhoncus.
```
[the remark this answers is not in the document]
Comment (konraddysput) [reply Backtrace-iOS/Resources/test.txt:1]: We use it in tests

**Review by konraddysput — APPROVED** [review]

**Comment by KrzaQ** [comment]
I don't have much to add above what Konrad said, given my lack of expertise. But I see a lot of commented out code/configs being committed. Maybe just remove it?

File: Backtrace.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{
+  "pins" : [
+    {
+      "identity" : "cwlcatchexception",
+      "kind" : "remoteSourceControl",
+      "location" : "https://github.com/mattgallagher/CwlCatchException.git",
```
[the remark this answers is not in the document]
Comment (melekr) [reply Backtrace.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:6]: yes it's a sub-dependency of Quick/Nimble

File: Backtrace.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      "state" : {
+        "revision" : "35f9e770f54ce62dd8526470f14c6e137cef3eea",
+        "version" : "2.1.1"
+      }
+    },
+    {
+      "identity" : "cwlpreconditiontesting",
```
[the remark this answers is not in the document]
Comment (melekr) [reply Backtrace.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:13]: sub-dependency of Quick/Nimble

File: Podfile (hunk 4 above)
Comment (melekr) [reply Podfile:7]: Will populate it again for Cocoa & spm merge,
what do you suggest we put here in the meantime?

File: scripts/test.sh (hunk 5 above)
Comment (melekr) [reply scripts/test.sh:11]: yes

**Pull request merged**